### PR TITLE
Ignore zero properties for MemoryInfo

### PR DESCRIPTION
**Pull request by SimonCropp**
WriteNumberIfNotZero since on OS that dont implement all props, thos props are stubbed out to zero
unknown

### CODE REVIEW

File: src/Sentry/Internal/MemoryInfo.cs [hunk 1]
```diff
@@ -81,23 +82,24 @@ internal sealed class MemoryInfo : IJsonSerializable
...
-            writer.WriteNumber("pause_time_percentage", PauseTimePercentage);
-            writer.WriteNumber("index", Index);
-            writer.WriteNumber("generation", Generation);
-            writer.WriteNumber("finalization_pending_count", FinalizationPendingCount);
+            writer.WriteNumberIfNotZero("total_committed_bytes", TotalCommittedBytes);
+            writer.WriteNumberIfNotZero("promoted_bytes", PromotedBytes);
+            writer.WriteNumberIfNotZero("pinned_objects_count", PinnedObjectsCount);
+            writer.WriteNumberIfNotZero("pause_time_percentage", PauseTimePercentage);
+            writer.WriteNumberIfNotZero("index", Index);
+            writer.WriteNumberIfNotZero("generation", Generation);
```
Comment (bruno-garcia) [inline src/Sentry/Internal/MemoryInfo.cs:101]: Gen0 would be `0` here? What does `generation` usually return?

Comment (bruno-garcia) [reply src/Sentry/Internal/MemoryInfo.cs:101]: Actually does it make sense to add generation at all? https://docs.microsoft.com/en-us/dotnet/api/system.gcmemoryinfo.generation?view=net-6.0#system-gcmemoryinfo-generation

We're not triggering GC collection here, right? We just want GC info and this doesn't seem to be relevant/helpful

File: src/Sentry/Internal/MemoryInfo.cs [hunk 2]
```diff
@@ -81,23 +82,24 @@ internal sealed class MemoryInfo : IJsonSerializable
...
+            writer.WriteNumberIfNotZero("finalization_pending_count", FinalizationPendingCount);
```
Comment (bruno-garcia) [inline src/Sentry/Internal/MemoryInfo.cs:102]: Would 0 be a valid value here? https://docs.microsoft.com/en-us/dotnet/api/system.gcmemoryinfo.finalizationpendingcount?view=net-6.0#system-gcmemoryinfo-finalizationpendingcount

**Review by bruno-garcia — COMMENTED** [review]
I wonder if there's some API we can call that will tell us whether `GetGCMemoryInfo` returns meaningful data or not. I mean: When do we get all zeros/defaults?

File: src/Sentry/Internal/MemoryInfo.cs (hunk 1 above)
Comment (SimonCropp) [reply src/Sentry/Internal/MemoryInfo.cs:101]: happy to remove Generation. but do u want that in diff PR with a diff release note?

**Comment by SimonCropp** [comment]
@bruno-garcia 

> I wonder if there's some API we can call that will tell us whether GetGCMemoryInfo returns meaningful data or not. I mean: When do we get all zeros/defaults?

not from my research

**Review by mattjohnsonpint — APPROVED** [review]

**Pull request merged**